test-requirements.txt

```diff
@@ -0,0 +1,3 @@
+pytest>=8.3.2
```

Comment:
Wondering if we can align the environment version with our current app runtime python packages for high fidelity testing https://src.dev.databricks.com/databricks-eng/universe@3cfd7ba4da9b32c458c0812898a482e03b127910/-/blob/apps/runtime/Dockerfile?L25-35